_posts/2021-02-23-OpenSSL.markdown

```diff
@@ -0,0 +1,94 @@
+- More recently, Gentoo, which used to offer LibreSSL as an option, has
+  [discontinued that
+  support](https://www.gentoo.org/support/news-items/2021-01-05-libressl-support-discontinued.html)
+  as well.
+
+For further context, LWN [covered the subject of LibreSSL on
+Linux](https://lwn.net/Articles/841664/) at the beginning of this year. The
+article also covers the attention and improvements that the OpenSSL project has
+received post-Heartbleed, which was one of the main reasons for Void's initial
+switch to LibreSSL.
+
+For an example of extra work caused by packages that expect OpenSSL, Void's
+version of Qt5 is heavily patched in order to properly support LibreSSL.
+Furthermore, the just released Qt6 would also need significant effort to be
+patched and maintained to use LibreSSL, without the possibility of such effort
+being upstreamed.
+
+At the same time, other software can have limited functionality or hit edge
+cases when using LibreSSL, due to not being thoroughly tested with it. One
+example is Python, which is limited in the ciphers it can offer to running
+programs, since it depends on what the SSL library provides - they are even
+[considering dropping any support for LibreSSL in the
+future](https://www.python.org/dev/peps/pep-0644/#libressl-support). Another
+example is OpenVPN, which was having connection issues with LibreSSL
+([void-linux/void-packages#23413](https://github.com/void-linux/void-packages/issues/23413)).
+This required us to switch the package to use Mbed TLS by default, which
+resulted in some limitations to the resulting package. With the switch to
+OpenSSL, the OpenVPN package will now be provided in the most widely deployed
+and tested configuration, so similar compatibility hiccups are unlikely to
+reappear.
+
+Unfortunately, this move has required us to drop some packages that rely on the
+OpenSSL 1.0.1 API; while LibreSSL maintains compatibility with this legacy API,
+modern OpenSSL has abandoned it.
+
+One great feature of LibreSSL not offered by OpenSSL is the `libtls` library,
+which aims to be a simple interface for communicating over the web safely, with
+sane defaults and hard to misuse. Some of the packages we ship depend on it, so
+work has been done to package a standalone version of it in
+[void-linux/void-packages#28732](https://github.com/void-linux/void-packages/pull/28732).
+
+The Void Linux team is grateful for the work of the OpenBSD community on
+[libressl-portable](https://github.com/libressl-portable/portable), we have
+greatly benefited from their work. Void continues to package other excellent
+OpenBSD software, including OpenSSH and signify. Signify is our tool of choice
+for signing live images, which can be obtained from our [download
+page](https://voidlinux.org/download/).
+
+In conclusion, we expect our switch to OpenSSL to ease maintenance overhead,
+provide the same reliable experience to users, and improve functionality in
+select packages.
```

Comment:
Very abrupt, needs another sentence after it to the effect of "look forward to the update landing some time after March 5th"

Reply:
I just copied your sentence :)

Reply:
you did, I thought this was going at the end of a larger paragraph, or at the start of one.

Reply:
I meant that I copied this new sentence onto that paragraph :p 

which was indeed already a sentence you had suggested previously